### ISSUE TITLE
[java] DataflowAnomalyAnalysis: DD false positive 

### ISSUE THREAD

**Issue by Reissner** [pmd/pmd#400]
Rule: [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis)

Code sample:

``` java
public class Test {
    public T addS(List<T> args) {
	T res = zeroS();//<res
	for (T arg : args) {
	    res = res.add(arg);
	}

	return res;
    }
}
```

yields false positive on variable res. 

Result:
`n/a:2:	Found 'UR'-anomaly for variable 'arg' (lines '2'-'5').`

**Comment by adangel** [comment]
The rule [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis) has been deprecated in favour of [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment), which is available with PMD 6.26.0.

Therefore this issue wont be fixed.

See also #2647.
